Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,247 @@
+					<failure message="PROGRAM.cbl:2 Use a program name that matches the source file name" type="WARNING">
+		WARNING: Use a program name that matches the source file name
+		Category: COBOL Code Review – Naming Conventions
+		File: /project/PROGRAM.cbl
+		Line: 2
+			  </failure>
+			</testcase>
+		  </testsuite>
+		</testsuites>
+*/
+
+Define Class(
+	"UtJunitXMLReporter",
+	Base Class( UtCollectingReporter ),
+
+/*	Function: generate_xml_tag
+        ---Prototype---
+        generate_xml_tag(<tag>, <content>, <properties>)
+        ---------------
+        takes in a tag as a string and content, puts the content into the tag in xml format
+      
+    Arguments:
+		tag: string: string that the tag should be
+		content: string: content that goes inside the string
+		properties: associative array: associative array with properties as keys and the values
+
+    Example:
+        This is an example
+        ---JSL---
+		test = generate_xml_tag("test", "content", ["prop"=>1]);
+
+		Expected Return:
+		<test prop="1">
+		  content
+		</test>
+        ---------
+*/
+	generate_xml_tag = Method({tag, content, properties},
+		xml_tag = "<" || tag;
+		props = properties << get keys;
+		vals = properties << get values;
+		For(i=1, i<=N Items(props), i++,
+			name = props[i];
+			If(Type(name) != "String",
+				Throw("property argument has a non string key found type:" || Type(name));
+			);
+			xml_tag ||= " " || name || "=\!"" || Char(vals[i]) || "\!"";
+		);
+		xml_tag ||= ">" || "\!n";
+		
+		If(content != "",
+			// replace all newline with doublespace then newline
+			Substitute Into(content, "\!n", "\!n  ");
+			xml_tag ||= "  " || content || "\!n";
+		);
+
+		xml_tag ||= "</" || tag || ">\!n";
+		Return(xml_tag);
+	);
+
+
+/*	Function: show_xml_report
+        ---Prototype---
+        show_xml_report(<xml_results>)
+        ---------------
+        takes a hash which should have test suites as keys, returns the JunitXML format
+      
+    Arguments:
+		xml_result: Associative Array: associative array where keys should be the test suites.
+		Values will be associative arrays with test cases as keys.
+		The values of those associative arrays will be failure messages with "failure" as the key.
+
+    Example:
+        This is an example
+        ---JSL---
+		xml_results = ["Example Tests" => 
+					    ["Test1" => 
+						  ["failure" =>
+							"Expected: add_nums(1, 1) equal to 3
+							But: was 2"
+						  ]
+						]
+					  ];
+		show_xml_report(xml_results);			
+
+		Expected Return:
+		<testsuites failures="1" id="20191011_175324" tests="1">
+			<testsuite failures="1" name="Example Tests" tests="1">
+				<testcase name="Test1">
+					<failure type="failure">
+						Expected: add_nums(1, 1) equal to 3
+						But: was 2
+					</failure>
+				</testcase>
+			</testsuite>
+		</testsuites>
+        ---------
+*/
+	show_xml_report = Method({xml_results},
```

Comment:
Maybe change the name of this method to `get_xml_report`? It does not do any showing, just returns a string.

Reply:
sounds good, I will make the change

Reply:
Done in commit 846e93f3706107d35a48fd74d83ad7285baa46c2